pages/admin/dashboard/builder.tsx

```diff
@@ -0,0 +1,268 @@
+                <Spacer />
+                <ButtonGroup variant="ghost">
+                    <IconButton aria-label="undo" icon={<IoIosUndo />} />
+                    <IconButton aria-label="redo" icon={<IoIosRedo />} />
+                    <IconButton aria-label="trash" icon={<IoTrash />} />
+                </ButtonGroup>
+            </Flex>
+            <Flex h="80vh">
+                <Box w={isSidebarOpen ? "60%" : "5%"} transition="all 0.25s">
+                    <Flex bg={isSidebarOpen ? "black" : "white"}>
+                        <Spacer />
+                        <Button
+                            variant={isSidebarOpen ? "solid" : "ghost"}
+                            colorScheme="blackAlpha"
+                            onClick={toggleSidebar}
+                        >
+                            {isSidebarOpen ? "<<" : ">>"}
+                        </Button>
+                    </Flex>
+                    {isSidebarOpen && (
+                        <Box>
+                            <Container maxW="70%" py={4}>
+                                <Stack spacing={2}>
+                                    <Heading>Section {slideNumber}</Heading>
+                                    <MarkdownEditor
+                                        value={editorText}
+                                        setValue={setEditorText}
+                                    />
+                                </Stack>
+                            </Container>
+                            <Container>
+                                <Stack spacing={10} direction="row">
+                                    <CheckboxComp
+                                        defaultIsChecked
+                                        text="Prev"
+                                        onChange={() => {
+                                            setPrevButton(!prevButton);
+                                        }}
+                                    />
+                                    <CheckboxComp
+                                        defaultIsChecked
+                                        text="Next"
+                                        onChange={() => {
+                                            setNextButton(!nextButton);
+                                        }}
+                                    />
+                                    <CheckboxComp
+                                        text="Save"
+                                        onChange={() => {
+                                            setSaveButton(!saveButton);
+                                        }}
+                                    />
+                                    <CheckboxComp
+                                        text="Print"
+                                        onChange={() => {
+                                            setPrintButton(!printButton);
+                                        }}
+                                    />
```

Comment:
If you have time, it would be ideal to convert this into a mapped array. It'll make it easy for us to add or take away buttons in the future and will make the code more concise as well :)

Reply:
Done!